javascript/ql/src/semmle/javascript/SSA.qll

```diff
@@ -521,6 +521,27 @@ class SsaExplicitDefinition extends SsaDefinition, TExplicitDef {
   ) {
     getDef().getLocation().hasLocationInfo(filepath, startline, startcolumn, endline, endcolumn)
   }
+
+  /**
+   * Gets the data flow node representing the incoming value assigned at this definition,
+   * if any.
+   */
+  DataFlow::Node getRhsNode() {
```

Comment:
This looks pretty similar to `defSourceNode` from `DataFlow.qll`. Could you comment on the differences?

Reply:
As discussed, the only difference seems to be the inclusion of default values from destructuring reads. I've updated the PR to use this instead. I'd like to deal with the ambiguous RHS by changing the dataflow graph to ensure a unique RHS node actually exists.